website_sale_product_contract_gift/static/src/js/website_sale.js

```diff
@@ -0,0 +1,29 @@
+odoo.define("website_sale_product_contract_gift.website_sale", function (require) {
+    "use strict";
+
+    var publicWidget = require("web.public.widget");
+
+    publicWidget.registry.websiteSaleCartGift = publicWidget.Widget.extend({
+        selector: ".oe_website_sale .oe_cart",
+        events: {
+            "click #gift_form_confirm_button": "_onClickConfirm",
+        },
+
+        // --------------------------------------------------------------------------
+        // Handlers
+        // --------------------------------------------------------------------------
+
+        /**
+         * @private
+         * @param {Event} ev
+         */
+        _onClickConfirm: function () {
+            var $form = $("form[name='date_for_gift']");
+            $form.submit();
+        },
+    });
+
+    return {
+        websiteSaleCartGift: publicWidget.registry.websiteSaleCart,
+    };
+});
```

Comment:
this still uses the old `define()` syntax and the [old class system](https://www.odoo.com/documentation/16.0/developer/reference/frontend/javascript_reference.html#class-system). this should be changed to use standard es6 modules and classes. please use the new syntax and rename the file to `website_sale.esm.js` (or better: `website_sale_cart_gift.esm.js`). something like this (not sure it will work as is):
```suggestion
/** @odoo-module **/

import {registry, Widget} from "web.public.widget";

export class WebsiteSaleCartGift extends Widget {
    _onClickConfirm() {
        const $form = $("form[name='date_for_gift']");
        $form.submit();
    }
}

WebsiteSaleCartGift.selector = ".oe_website_sale .oe_cart";
WebsiteSaleCartGift.events = {
    "click #gift_form_confirm_button": "_onClickConfirm",
};
registry.websiteSaleCartGift = WebsiteSaleCartGift;
```

Reply:
Thanks a lot. Still difficult to me to translate to new framework when I base my work on old-style examples.

Reply:
done.

Reply:
the transition from `odoo.define()` and `require` to the es6 syntax is now done, thanks! the code is still using the old `.extend()` method instead of the `extends` syntax of es6, though. did you try the suggestion above?

Reply:
I tried the suggested code above, but it does not work. The method was never triggered. I get inspired by the code from odoo 18.0 for the solution I proposed.

Reply:
ok, let’s leave it like this (at least for now) then.